src/program/lwaftr/tests/subcommands/run_nohw_test.py

```diff
@@ -0,0 +1,77 @@
+"""
+Test the "snabb lwaftr run_nohw" subcommand.
+"""
+
+import unittest
+
+from random import randint
+from subprocess import call, check_call
+from test_env import DATA_DIR, SNABB_CMD, BaseTestCase
+
+class TestRun(BaseTestCase):
+
+    program = [
+        str(SNABB_CMD), 'lwaftr', 'run_nohw',
+    ]
+    cmd_args = {
+        '--duration': '1',
+        '--bench-file': '/dev/null',
+        '--conf': str(DATA_DIR / 'icmp_on_fail.conf'),
+        '--inet-if': '',
+        '--b4-if': '',
+    }
+    veths = []
+
+    @classmethod
+    def setUpClass(cls):
+        cls.create_veth_pair()
+
+    @classmethod
+    def create_veth_pair(cls):
+        veth0 = cls.random_veth_name()
+        veth1 = cls.random_veth_name()
+
+        # Create veth pair.
+        check_call(('ip', 'link', 'add', veth0, 'type', 'veth', 'peer', \
+            'name', veth1))
+
+        # Set interfaces up.
+        check_call(('ip', 'link', 'set', veth0, 'up'))
+        check_call(('ip', 'link', 'set', veth1, 'up'))
+
+        # Add interface names to class.
+        cls.veths.append(veth0)
+        cls.veths.append(veth1)
+
+    @classmethod
+    def random_veth_name(cls):
+        return 'veth%s' % randint(10000, 999999)
+
+    def test_run_nohw(self):
+        self.execute_run_test(self.cmd_args)
+
+    def execute_run_test(self, cmd_args):
+        self.cmd_args['--inet-if'] = self.veths[0]
+        self.cmd_args['--b4-if'] = self.veths[1]
+        output = self.run_cmd(self.build_cmd())
+        self.assertIn(b'link report', output,
+            b'\n'.join((b'OUTPUT', output)))
+
+    def build_cmd(self):
+        result = self.program
+        for item in self.cmd_args.items():
+            for each in item:
+                result.append(each)
+        return result
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.remove_veths()
+
+    @classmethod
+    def remove_veths(cls):
+        for i in range(0, len(cls.veths), 2):
+            check_call(('ip', 'link', 'delete', cls.veths[i]))
```

Comment:
To delete a veth pair is just necessary to delete one of the ends, otherwise an error is returned when trying to remove both. This was the reason why the loop iterated even numbers. In any case, as there's only two elements in the array as the test is now I removed the loop completely and just remove veth[0].